<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>RNA-seq introduction</title>
    <meta charset="utf-8" />
    <meta name="author" content="Mikhail Dozmorov" />
    <meta name="date" content="2025-09-22" />
    <script src="libs/header-attrs-2.29/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# RNA-seq introduction
]
.author[
### Mikhail Dozmorov
]
.institute[
### Virginia Commonwealth University
]
.date[
### 2025-09-22
]

---




&lt;!-- HTML style block --&gt;
&lt;style&gt;
.large { font-size: 130%; }
.small { font-size: 70%; }
.tiny { font-size: 40%; }
&lt;/style&gt;

## Eukaryotic gene structure

&lt;img src="img/gene_structure.png" width="850px" style="display: block; margin: auto;" /&gt;

---
## Gene expression

&lt;img src="img/central_dogma1.png" width="850px" style="display: block; margin: auto;" /&gt;

---
## Gene expression

&lt;img src="img/central_dogma2.png" width="850px" style="display: block; margin: auto;" /&gt;

---
## What is RNA sequencing?

- Massive parallel sequencing to **characterize and quantify transcriptomes** (all actively transcribed genes)

- Detection of **differential gene expression**

- **Transcriptome reconstruction**, identification of **new transcripts**

- Detection of **alternative splicing events**

- Detection of **structural variants**, e.g., fusion transcripts

- **Allele-specific** gene expression measurements

- **Mutation analysis** – presence of genomic mutations and their effect on gene expression

.small[ RNA-seq analysis techniques: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&amp;id=info:doi/10.1371/journal.pcbi.1004393.s003 ]

---
## Sequencing technologies

Commercially available

- **Illumina/Solexa** - short reads, sequencing-by-synthesis

- **Life Technologies Ion Torrent/Proton** - short reads, Ion Semiconductor sequencing

- **Pacific Biosciences** - long reads, Single Molecule Real Time sequencing

Experimental

- **Nanopore sequencing** - continuous sequencing (very long reads), fluctuations of the ionic current from nucleotides passing through the nanopore

&lt;!--
## Cap Analysis of Gene Expression (CAGE)

- The first 20 to 21 nucleotides at the 5' ends of capped mRNAs are extracted by a combination of cap trapping and cleavage by restriction enzyme _MmeI_.
- The "CAGE tags" then sequenced and mapped to the genome
- Produces a genome-wide picture of transcription start sites at a base-pair resolution
- "Promoterome" generated by FANTOM project

.small[ Balwierz, Piotr J., Piero Carninci, Carsten O. Daub, Jun Kawai, Yoshihide Hayashizaki, Werner Van Belle, Christian Beisel, and Erik van Nimwegen. “Methods for Analyzing Deep Sequencing Expression Data: Constructing the Human and Mouse Promoterome with DeepCAGE Data.” Genome Biology 10, no. 7 (2009): R79. doi:10.1186/gb-2009-10-7-r79. https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-7-r79
--&gt;

---
## Overview of RNA sequencing technology

&lt;img src="img/rna-seq_tech.png" width="800px" style="display: block; margin: auto;" /&gt;

.small[ Source: https://doi.org/10.1371/journal.pcbi.1004393 ]

---
## Advantage of RNA-Seq

**Much richer information beyond quantitation**

- Boundary of gene transcripts: both 5' and 3' end, to nucleotide level 

- Alternative exon usage, novel splicing junction detection

- SNP/indel discovery in transcripts: both coding and UTRs

- Allele specific expression: critical in imprinting, cancer

- Novel gene and exon discovery enabled

---
## Advantage of RNA-Seq

**Flexibiliry at quantitation**

- Unlimited dynamic range: by increasing depth as needed

- Higher specificity and accuracy: digital counts of transcript copies, very low background noise

- Higher sensitivity: more transcripts and more differential genes detected

- Re-analysis easily done by computation, as gene annotation keeps evolving 

- _De novo_ assembly possible, not relying on reference genome sequence 

&lt;!--
## SEQC RNA-seq quality control.

- The third phase of the MicroArray Quality Control (MAQC) project, also known as the Sequencing Quality Control (SEQC) project, aims to assess the technical performance of high-throughput sequencing platforms by generating benchmarking data sets
- The design includes four different sample types, namely samples A, B, C and D
    - Sample A is Stratagene’s universal human reference (UHR) RNA
    - Sample B is Ambion’s human brain reference RNA
    - Samples C and D are mixes of samples A and B, in a 3:1 and 1:3 ratio, respectively
- The four reference samples were sent to several sequencing centers around the world and sequenced using different platforms
- Found good reproducibility across sites and platforms

.small[ SEQC/MAQC-III Consortium. A comprehensive assessment of RNA-seq accuracy, reproducibility and information content by the Sequencing Quality Control Consortium. Nat. Biotechnol. doi:10.1038/nbt.2957 (24 August 2014).
--&gt;

---
## RNA-Seq Limitations

Quantitation influenced by many confounding factors  

- "Sequenceability" - varying across genomic regions, local GC content and structure-related

- Varying length of gene transcripts and exons

- Bias in read ends due to reverse transcription, subtle but consistent 

- Varying extent of PCR amplification artifacts

- Effect of RNA degradation in the real world

- Computational bias in aligning reads to genome due to aligners
    
---
## RNA-Seq Limitations

SNP discovery in RNA-seq is more challenging than in DNA  

- Varying levels of coverage depth
- False discovery around splicing junctions due to incorrect mapping
    
_De novo_ assembly of transcripts without genome sequence: computationally intensive but possible, technical improvements will help  

- Longer read length
- Lower error rate
- More uniform nucleotide coverage of transcripts - more equalized transcript abundance

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
  "ratio": "16:9",
  "highlightStyle": "github",
  "highlightLines": true,
  "countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
