---
title: "BIOS658, Fall 2025"
output:
  word_document: default
  html_document: default
# date: "2024-09-17"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Course Title: Statistical Methods for High-throughput Genomics Data I

Instructor: Mikhail Dozmorov  
Department: Biostatistics, VCU  
Credits: 3   
Duration: 15 weeks (2 lectures per week, 1 hour 20 minutes each)

## Course Overview

<!-- This course introduces deep learning concepts using R, with hands-on exercises. The focus will be on implementing deep learning models and exploring various architectures using the "Deep Learning with R" textbook by François Chollet and J. J. Allaire. The course will cover both foundational and advanced topics in deep learning, including computer vision, natural language processing, and generative models. -->

Welcome to BIOS 658. This course is the first part of the Genomics curriculum, aimed to introduce core principles of Data Science, Genomics, Bioinformatics and Biostatistics. This is a blended course that combines in-class learning (lectures, labs) with self-directed activities.

Bioinformatics is interdisciplinary computer science, requiring a knowledge base in biology, technology, and statistics. The operational definition of ‘bioinformatics’ for this course is "The application of biotechnology and statistical methods to the study of biological problems." The biological problems on which we will focus are primarily gene expression studies conducted using RNA-seq technology.

The class will be conducted in person and include lecture and coding parts. Course material will be publicly available. The syllabus is subject to change. Observe the [VCU Honor Pledge](https://students.vcu.edu/studentconduct/vcu-honor-system/academic-misconduct-/honor-pledge/) in any class- and homework activities.

## Prerequisites

BIOS 524; and BIOS 544 or BIOS 654 (see [Bulletin](https://bulletin.vcu.edu/azcourses/bios/)), or as permitted by instructor. 

<!--- Book
    - [**Deep learning with R**](https://www.manning.com/books/deep-learning-with-r-second-edition) by François Chollet (the creator of Keras) with J. J. Allaire (the founder of RStudio and the author of the R interfaces to Keras and TensorFlow) -->

<!--    - [The Deep Learning textbook](https://www.deeplearningbook.org/) by Ian Goodfellow, Yoshua Bengio and Aaron Courville
    - [MIT Introduction to Deep Learning | 6.S191](https://www.youtube.com/watch?v=njKP3FqW3Sk&list=PLtBw6njQRU-rwp5__7C0oIVt26ZgjG9NI) - MIT video course by Alexander Amini, Ava Soleimani, and guests. Dense and informative ~45min lectures covering various topics of deep learning. [introtodeeplearning.com](http://introtodeeplearning.com/) - course web site with slides, video, and other material
    - [Machine learning and deep learning resources](https://github.com/mdozmorov/MachineLearning_notes) - a collection of references for further studies
- Code
    - [**R notebooks for the code samples of the book "Deep Learning with R"**](https://github.com/jjallaire/deep-learning-with-r-notebooks) and the [second edition code](https://github.com/t-kalinowski/deep-learning-with-R-2nd-edition-code)
    - [Deep Learning with Keras and TensorFlow in R Workflow](https://github.com/rstudio-conf-2020/dl-keras-tf) - RStudio Conference 2020 workshop by Brad Boehmke
-->

<!--- Skills
    - Working knowledge of R, familiarity with RStudio programming environment, command line, GitHub (BIOS524)
    - Basic linear algebra: vectors, matrices, determinants
    - Simple calculus: derivatives, integrals, gradients
    - Some probability theory: probability, random variables, distributions
    - Basic statistics knowledge: descriptive statistics, estimators.
    - (Linear) modeling
-->

- Hardware
    - A laptop, Mac or Linux OSs are recommended.

- Software
    - R for [**Windows**](http://cran.r-project.org/bin/windows/base/) or [**Mac**](http://cran.r-project.org/bin/macosx/). Review [**Getting Used to R, RStudio, and RMarkdown**](https://ismayc.github.io/rbasics-book/) book, if necessary
    - [**RStudio Desktop**](https://www.rstudio.com/products/rstudio/download/)

<!--    - [**Git**](https://git-scm.com/downloads)
    - A text editor ([**Notepad++**](https://notepad-plus-plus.org/) on Windows, or [**Sublime text**](https://www.sublimetext.com/) on any platform)
    - Optional: Docker for [**Windows**](https://hub.docker.com/editions/community/docker-ce-desktop-windows/) or [**Mac**](https://hub.docker.com/editions/community/docker-ce-desktop-mac/) is recommended
    - Windows only: [**Git Bash**](https://git-for-windows.github.io/) or [**Cygwin**](http://www.cygwin.com/)
    - Windows only: [**Rtools**](https://cran.r-project.org/bin/windows/Rtools/)
    - [**Windows-specific instructions on installing Keras and TensorFlow**](tensorflow.html)
-->

### Course Objectives

* Gain insight into biological principles and the RNA-seq technology, normalization, and expression summary methods
* Learn practical Exploratory Data Analysis, visualization, and quality control
* Critically evaluate ant interpret statistical methods used in genomics data analysis
* Apply supervised and unsupervised methods to genomics data
* Assess statistical significance when multiple hypothesis tests are performed, such as in the analysis of differential gene expression
* Interpret biological findings obtained from statistical tests

At the conclusion of the course, students will be able to collect, analyze and interpret genomics data using R programming environment

### Tentative schedule 

- Reproducible research using R, Rmarkdown, tidyverse, GitHub
- Genomics overview, Biological background
- RNA-seq technology
- Bioconductor overview
- Exploratory Data Analysis (EDA), Quality control
- Normalization, Expression summarization
- Differential expression
- Multiple hypothesis testing
- Clustering, dimensionaliry reduction
- Gene Ontology, Pathway and Functional enrichment
- Networks
- Machine learning

## Grading

- Assignments (60%): Regular homework assignments and the final project
- Participation (40%): Attendance and active participation in class discussions and labs
